lte/gateway/c/session_manager/SessionStore.cpp

```diff
@@ -148,9 +150,16 @@ SessionMap SessionStore::read_sessions_for_deletion(const SessionRead& req) {
   return session_map;
 }
 
+// maybe add shard id upon session creations!
 bool SessionStore::create_sessions(
     const std::string& subscriber_id, SessionVector sessions) {
-  auto session_map           = SessionMap{};
+  auto session_map = SessionMap{};
+  int shard_id     = ue_shards_->add_ue();
+  for (size_t i = 0; i < sessions.size(); i++) {
+    sessions[i]->set_shard_id(shard_id);
+  }
+  MLOG(MINFO) << "UE with IMSI: " << subscriber_id
+              << " and shard_id: " << shard_id;
```

Comment:
Note this breaks abstraction of Session Store. Session store should be dealing with storage of session, not defining fields. 

One place it could be added is on [update_tunnel_ids](http://github.com/magma/magma/blob/7441d3f1467d142e0f01f101c0e5c0a7a1228592/lte/gateway/c/session_manager/LocalEnforcer.cpp#L1023-L1023). That functions does many changes on sessions. 

This should be added before `handle_session_activate_rule_updates`, that is where the rules are sent to pipelined to be installed (and pipelined will need to know the shard id on activation)